### PR TITLE
preserves original options in `_cookiecutter`

### LINKED ISSUE #1873: getting a parameters original choices
* Cookiecutter version: 2.1.1
* Template project url: github.com/kjaymiller/cookiecutter_relecloud
* Python version: Python 3.11.0
* Operating System: MacOS Ventura - 13.4.1 (22F82)

### Description:

If my cookiecutter.json has:

```json
{
    "cc_options": [1, 2, 3, 4]
}
```
is there a way to get all of the options.

### What I've run:

```jinja
{{cookiecutter.json}}
```

```jinja
{{cookiecutter.cc_options}}
```

### Expected outcome

I would like some attibute that would give me my original options.  Perhaps if you call the value with an underscore in front.

```jinja
{{cookiecutter._cc_options}} >>> [1,2,3,4]
```

another option could be to have the original values preserved in a variable

```jinja
{{cookiecutter.source_values.cc_options}} >>> [1,2,3,4]
```

### PR DESCRIPTION
**Pull request by kjaymiller**
fixes #1873 

Change to `cookiecutter.main.py` in `cookiecutter()` that saves all the original context key:value pairs in `context['_cookiecutter']` before overwriting them in `context['cookiecutter']`.

### CODE REVIEW

File: tests/fake-repo-tmpl-_cookiecutter/cookiecutter.json
Comment (kjaymiller) [inline tests/fake-repo-tmpl-_cookiecutter/cookiecutter.json:1]: created a new fake-repo directory to add custom key:value pairs to ensure that lists, and dictionaries retained all their values.

File: tests/test_cli.py
Comment (kjaymiller) [inline tests/test_cli.py:1]: modified the cli echo error output test to no longer rely on exact JSON serialization as `result.output` was flaky and now also contained `_cookiecutter`

**Review by ericof — APPROVED** [review]
Thanks for your contribution

**Comment by noklam** [comment]
Is it possible to reference this `_cookiecutter` variable? I tried `{{ _cookiecutter.var }}` and seems like I get the same thing.

**Pull request merged**